src/MockClassifier.Api/Controllers/InstitutionController.cs

```diff
@@ -15,20 +28,30 @@ public class InstitutionController : ControllerBase
         [HttpPost]
         public AcceptedResult Post([FromBody] MessagesInput messages)
         {
-            var minsitries = new List<string>();
-
-            // TO DO
-            // Enumerate each string in messages and check for
-            // 1) Does the string match one of the pre-defined phrases? If so, work out the corresponding ministry and put it into minsitries
-            // 2) Does the string include one of more of the pre-defined tokens? If so, get the corresponding minsitries and put them into minsitries
-
-            foreach (var minsitry in minsitries)
+            foreach( var message in messages.Messages)
             {
-                // TO DO
-                // Invoke the DMR call back service to put a message on DMR for each misnitry
-            }
+                List<string> ministries = _naturalLanguageService.Classify(message).ToList();
+                ministries = ministries.Concat(_tokenService.Classify(message).ToList()).ToList();
 
+                foreach (var ministry in ministries)
+                {
+                     _dmrService.RecordRequest(GetDmrRequest(message, ministry, messages.CallbackUri));
+                }
+            }
             return Accepted();
         }
+
+        private static DmrRequest GetDmrRequest(string message, string ministry, string callbackUri)
```

Comment:
I was wondering whether we can move this logic into the DmrRequest object in file: DmrRequest.cs?